e2e/singleserver_singleagent_test.go

```diff
@@ -0,0 +1,130 @@
+package e2e
+
+import (
+	"context"
+	"fmt"
+	"net/http"
+	"strconv"
+	"testing"
+	"time"
+
+	"github.com/prometheus/common/expfmt"
+	"sigs.k8s.io/e2e-framework/klient/wait"
+	"sigs.k8s.io/e2e-framework/klient/wait/conditions"
+	"sigs.k8s.io/e2e-framework/pkg/envconf"
+	"sigs.k8s.io/e2e-framework/pkg/features"
+)
+
+func TestSingleServer_SingleAgent_StaticCount(t *testing.T) {
+	serviceHost := "konnectivity-server.kube-system.svc.cluster.local"
+	adminPort := 8093
+
+	serverDeploymentCfg := DeploymentConfig{
+		Replicas: 1,
+		Image:    *serverImage,
+		Args: []KeyValue{
+			{Key: "log-file", Value: "/var/log/konnectivity-server.log"},
+			{Key: "logtostderr", Value: "true"},
+			{Key: "log-file-max-size", Value: "0"},
+			{Key: "uds-name", Value: "/etc/kubernetes/konnectivity-server/konnectivity-server.socket"},
+			{Key: "delete-existing-uds-file"},
+			{Key: "cluster-cert", Value: "/etc/kubernetes/pki/apiserver.crt"},
+			{Key: "cluster-key", Value: "/etc/kubernetes/pki/apiserver.key"},
+			{Key: "server-port", Value: "8090"},
+			{Key: "agent-port", Value: "8091"},
+			{Key: "health-port", Value: "8092"},
+			{Key: "admin-port", Value: strconv.Itoa(adminPort)},
+			{Key: "keepalive-time", Value: "1h"},
+			{Key: "mode", Value: "grpc"},
```

Comment:
we should also test e2e for http-connect mode.

Perhaps this can be retrieved as environment variable and the e2e github workflow run twice once with grpc and second time with http-connect  ?

similar to how you have done for k8s matrix

Reply:
Added in the latest commit.